packages/ref-metrics-esmvaltool/src/ref_metrics_esmvaltool/recipe.py

```diff
@@ -0,0 +1,209 @@
+from __future__ import annotations
+
+import subprocess
+from pathlib import Path
+from typing import TYPE_CHECKING, Any
+
+import pkg_resources
+import pooch  # type: ignore[import-untyped]
+from ref_core.datasets import SourceDatasetType
+from ref_core.metrics import MetricExecutionDefinition
+from ruamel.yaml import YAML
+
+if TYPE_CHECKING:
+    import pandas as pd
+
+yaml = YAML()
+
+FACETS = {
+    "CMIP6": {
+        "dataset": "source_id",
+        "ensemble": "member_id",
+        "exp": "experiment_id",
+        "grid": "grid_label",
+        "mip": "table_id",
+        "short_name": "variable_id",
+    },
+}
+
+
+def as_isodate(timestamp: pd._libs.tslibs.timestamps.Timestamp) -> str:
```

Comment:
Does the top-level `pd.Timestamp` typehint not work or `pd.core.api.Timestamp`?

Reply:
Changed in b3897fd. I'm not very familiar with `pandas`, so I just looked at the runtime type to figure out what it was.